### DISCUSSION TITLE
Scaling issues

### DISCUSSION THREAD

**Discussion by LiuPengPeter** [idaholab/moose#16365] [Q&A Modules: Phase field] [answered]
Hello everyone,

In my phase field module, the value of material parameters are in different orders of magnitude. I try to scale variables and material parameters.

I have some questions about scaling. For example:
```
[Variables]
  [./**eta**]
    order = FIRST
    family = LAGRANGE
    **scaling= 1.0e5**
  [../]
```
Does it means that the variable eta becomes "eta*1.0e5" in all of the process of simulation and return "eta" without "1.0e5"
 in the final result?

Thank you a lot!

**Comment by mangerij** [comment, marked as answer]
It means the residual corresponding to eta is scaled by 1e5

On Dec 2, 2020 9:25 AM, LiuPengPeter <notifications@github.com> wrote:

*Message sent from a system outside of UConn.*


Hello everyone,

In my phase field module, the value of material parameters are in different orders of magnitude. I try to scale variables and material parameters.

I have some questions about scaling. For example:

[Variables]
  [./**eta**]
    order = FIRST
    family = LAGRANGE
    **scaling= 1.0e5**
  [../]


Does it means that the variable eta becomes "eta*1.0e5" in all of the process of simulation and return "eta" without "1.0e5"
in the final result?

Thank you a lot!

**Reply by LiuPengPeter** [reply]
Thanks for your help. Could you explain more details?

If there are R1, R2 and R3 whose variable is "V", and "eta" is just the coupled variable.
R1 = V + eta^2 + eta
R2 = partdiff{**eta**}{**x**}
R3 = partdiff{**V**}{**eta**}

I think that should be:
R1*(1.0e5)^2 + R2*1.0e5 +R3*(1/1.0e5).

Right?

**Reply by mangerij** [reply]
If you use 

```
[Debug]
  show_var_residual_norms = true
[]
```
in your input file, you will see the influence of using scaling for the residuals. Remember the residual is the total sum of all the kernel contributions for that variable. So if you have only two variables, you should have only two residuals (one each).

**Reply by LiuPengPeter** [reply]
I just want to take an example. What I care is that how the scaling factor influence the residual of the kernel. Could you tell me about the detail of scaling factor?

**Reply by GiudGiud** [reply]
We work with the residual of the variable. A single kernel does not (usually) define the whole PDE that rules the variable. 
The scaling factor is applied on the residual of the variable. 

Some kernels' residual may be scaled, for particular use cases, but that's not what you are doing here.
